Submitty/Submitty
# [Feature:InstructorUI] Add regrade buttons

### Please check if the PR fulfills these requirements:


### What is the current behavior?
Currently in order to regrade assignments, the instructor must do it from the command line with

sudo /usr/local/submitty/bin/regrade.py /var/local/submitty/courses/{{ semester}}/{{course}}/submissions/{{gradeable_id}}

The documentation for these commands are under instructor->assignment preparation->automated grading->batch regrade 
When any of the regrade commands are run, a file/files are created in /var/local/submitty/to_be_graded_queue

in order to see them there you must first stop your shipper and worker or they'll be graded right away.

Use:
-sudo systemctl stop submitty_autograding_shipper 
-sudo systemctl stop submitty_autograding_worker

An example of the contents of the file created may look like this
<img width="289" alt="pr10_1" src="https://user-images.githubusercontent.com/66340271/122287822-5ed49e80-cebf-11eb-9207-29dbc72d4497.PNG">

Once the file is created, the worker and shipper take care of the rest and the actual grading.

### What is the new behavior?
There are now multiple buttons available to the instructor (or TA's with instructor permissions) on the grading interface that allow them to regrade assignments.  
There are 4 options for regrading:
regrade active version - regrade the active version for one selected student who submitted a certain gradeable
regrade all versions - regrade every version for one selected student who submitted a certain gradeable
regrade all students active versions - regrade the active version for every student who submitted a certain gradeable
regrade all students all versions - regrade every version for every student who submitted a certain gradeable
<img width="531" alt="pr10_2" src="https://user-images.githubusercontent.com/66340271/122291015-56ca2e00-cec2-11eb-9997-870c684e0633.PNG">
<img width="735" alt="pr10_3" src="https://user-images.githubusercontent.com/66340271/122291024-59c51e80-cec2-11eb-81be-39d4af9b646d.PNG">

These buttons have the same functionality as the command line option. They add the properly formatted files to the to_be_graded_queue directory so that the worker and shipper can process them properly.

### Other information?
Regrading is currently not available for assignments that use submissions from github because of known minor bugs in the way that those submissions are set up so the buttons are hidden for those assignments.

I realize the split buttons aren't the prettiest but the bootstrap ones won't render properly as of now #6618 and we still decided that we split buttons are the best option for this case.

To test:
1. First use the commands I listed above to manually regrade assignments and look at the files created in the queue
2. Now use the buttons and compare the files for all sorts of different gradeables
3. Change the autograding configuration for an assignment and see the results after using the regrade buttons
